server/src/main/java/org/opensearch/index/compositeindex/datacube/DimensionDataType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Represents the type of comparison to be performed on a dimension.
```

Comment:
nit: please update the javadoc - 'represents the data type of the dimension value'